jl4/tests/Main.hs

```diff
@@ -94,20 +94,34 @@ jl4ExactPrintGolden dir inputFile = do
 
 jl4NlgAnnotationsGolden :: Bool -> String -> FilePath -> IO (Golden Text)
 jl4NlgAnnotationsGolden isOk dir inputFile = do
-  input <- Text.readFile inputFile
-  output_ <- case execProgramParser (toNormalizedUri $ filePathToUri inputFile) input of
-    Left _err -> pure "Failed to parse"
-    Right (prog, warns) -> do
-      null warns `shouldBe` isOk
-      pure $ prettyNlgOutput prog warns
+  (errs, moutput) <- oneshotL4ActionAndErrors inputFile \nfp -> do
+    let uri = normalizedFilePathToUri nfp
+    _ <- Shake.addVirtualFileFromFS nfp
+    Shake.use Rules.SuccessfulTypeCheck uri
+  let output_ = case moutput of
+        Nothing -> "Failed to typecheck\n"
```

Comment:
this is probably where the issue comes from that the parsing failed changed to typecheck failed: you need to actually check that the parsing rule failed explicitly (this is done in this module already, so you can copy that, I guess :))

Reply:
I don't really care about the parsing any more, since NLG annotations need to be type checked as well. Thus, I improved the message, but intend to keep it like this for now.